File: 프로그래머스/1/12926. 시저 암호/시저 암호.java
class Solution {
    public String solution(String s, int n) {

        StringBuilder sb = new StringBuilder();
        char[] cs = s.toCharArray();

        for (char c : cs) {

            if ('a' <= c && c <= 'z' || 'A' <= c && c <= 'Z') {

                if (c + n > 'z') {
                    sb.append((char)(c + n - 26));
                    continue;
                } else if (c + n > 'Z' && c <= 'Z') {
                    sb.append((char)(c + n - 26));
                    continue;
                }
                sb.append((char)(c + n));
            } else {
                sb.append(c);
            }
        }

        return sb.toString();
    }
}

File: Kotlin/프로그래머스/1/12926. 시저 암호/시저 암호.kt
class Solution {
    fun solution(s: String, n: Int): String {

        val sb = StringBuilder()

        for (c in s) {
            if (c != ' ') {

                sb.append(appending(c, n))
                continue
            }
            sb.append(" ")
        }

        return sb.toString()
    }

    private fun appending(c: Char, i: Int): Char {

        val newChar = c + i

        if (c in 'A'..'Z') {
            if (newChar > 'Z') {
                return newChar - 26
            }
        }

        if (c in 'a'..'z') {
            if (newChar > 'z') {
                return newChar - 26
            }
        }
        return newChar

    }
}
